src/constants.rs

```diff
@@ -382,12 +382,32 @@ impl SoundType {
 
 /// A uinput event published by the kernel into the events stream for uinput devices.
 #[derive(Copy, Clone, PartialEq, Eq)]
-pub struct UInputEventType(pub u16);
+pub struct UInputType(pub u16);
 
 evdev_enum!(
-    UInputEventType,
+    UInputType,
     /// The virtual uinput device is uploading a force feedback effect.
     UI_FF_UPLOAD = 1,
     /// The virtual uinput device is erasing a force feedback event.
     UI_FF_ERASE = 2,
 );
+
+// some more structs without any constats. They are only there to
+// porvide a consitatnt type system and simple code generation.
+#[cfg(feature = "serde")]
```

Comment:
👀

Reply:
In the `InputEventKind` enum there are a few variants which didn't follow the common pattern:
```rs
pub enum InputEventKind {
    Synchronization(Synchronization),
    Key(Key),
    RelAxis(RelativeAxisType),
    AbsAxis(AbsoluteAxisType),
    Misc(MiscType),
    Switch(SwitchType),
    Led(LedType),
    Sound(SoundType),
    ForceFeedback(u16),
    ForceFeedbackStatus(u16),
    UInput(u16),
    Other,
}
```

So I created a few newtypes like the `PowerType` and the `OtherType`, so they all follow the same pattern. 
That comment references generally the stuff below. I can move it I you feel that would be clearer.